### DISCUSSION TITLE
InstrumentedRecordInterceptor does not implement afterRecord() method

### DISCUSSION THREAD

**Discussion by phillipdriver** [open-telemetry/opentelemetry-java-instrumentation#10934] [Q&A]
Is there a good reason as to why [InstrumentedRecordInterceptor](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-kafka-2.7/library/src/main/java/io/opentelemetry/instrumentation/spring/kafka/v2_7/InstrumentedRecordInterceptor.java) doesn't implement Spring Kafka's `RecordInterceptor#afterRecord()` method?

I've written a simple Spring Boot Kafka consumer app that implements and registers a custom `RecordInterceptor`, however, when instrumenting the app using the Open Telemetry java agent ([opentelemetry-javaagent.jar](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/latest/download/opentelemetry-javaagent.jar)), because `InstrumentedRecordInterceptor` doesn't implement `afterRecord()` the default `RecordInterceptor#afterRecord()` method is called (which does nothing) rather than calling back into my custom `RecordInterceptor`.

This seems a bit odd?

**Comment by laurit** [comment]
I create https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10935 to address this

**Comment by phillipdriver** [comment]
Just to add a bit of an answer to my question, seems Spring added a few new methods to the `RecordInterceptor` interface in v2.8.0, `afterRecord()` being one of them. OpenTelemetry were supporting v2.7.0, hence not delegating for these new methods. @laurit has raised a PR to address this.
